shared-libs/transitions/src/transitions/generate_patient_id_on_people.js

```diff
@@ -1,21 +1,34 @@
 const config = require('../config');
 const transitionUtils = require('./utils');
+const contactTypeUtils = require('@medic/contact-types-utils');
 
+// As of 3.8.0, generates a shortcode for every configured contact type.
 module.exports = {
   filter: doc => {
-    if (doc.patient_id) {
-      // already has a patient id
+    const contactType = contactTypeUtils.getContactType(config.getAll(), doc);
+    if (!contactType) {
       return;
     }
-    const typeId = (doc.type === 'contact' && doc.contact_type) || doc.type;
-    const contactTypes = config.get('contact_types') || [];
-    const type = contactTypes.find(type => type.id === typeId);
-    return type && type.person;
+
+    if (contactTypeUtils.isPersonType(contactType) && doc.patient_id) {
+      return; // person type that already had patient_id
+    }
+
+    if (contactTypeUtils.isPlaceType(contactType) && doc.place_id) {
+      return; // contact type that already has place_id
+    }
+
+    return true;
   },
   onMatch: change => {
+    const isPerson = contactTypeUtils.isPerson(config.getAll(), change.doc);
```

Comment:
Move this down into the `.then` online 28 so it's scoped as tightly as possible.

Reply:
done